HTML Element page template - update for browser-compat

hamishwillee opened the pull request:
This updates the [HTML element page template](https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/HTML_element_page_template) for improved front-matter information (in particular browser compat) and generally for best-practice.

hamishwillee committed:
HTML Element page template - update to use better front matter description

teoli2003 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
-	<li>
-	<div>\{{htmlref}} — this generates a suitable HTML sidebar, depending on what tags are included on the page.</div>
-	</li>
+	<li><code>\{{draft()}}</code> — this generates a <strong>Draft</strong> banner that indicates that the page is not yet complete, and should only be removed when the first draft of the page is completely finished. After it is ready to be published, you can remove this.</li>
```
Do we want to continue recommanding the {{Draft}} macro? I think that with the github contribution process, it is no more needed: draft pages would not land (yet).

A kind of remnant of the wiki era.

hamishwillee replied:
That I do not know. It is a very good question. If it comes to a vote, then I say "yes", nothing should be submitted if it isn't already past basic draft.

hamishwillee replied:
@teoli2003 Also consider https://github.com/mdn/yari/discussions/4071

Elchi3 replied:
I agree to remove this.

hamishwillee replied:
I have deleted this from this doc, and all the other templates.

Elchi3 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
 
-<p>The <em>Slug</em> (the last segment at the end of the URL) should be filled in as <em>NameOfTheElement</em>, so <code>&lt;video&gt;</code>'s slug is <em>video</em>.</p>
+<pre>---
+title: '&lt;NameOfTheElement&gt;: The NameOfTheElement Embed element'
```
```suggestion
title: '&lt;NameOfTheElement&gt;: The NameOfTheElement element'
```

Elchi3 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
+<h3 id="Page_front_matter">Page front matter</h3>
 
-<p>An HTML element page should have a <em>title</em> of <em>"&lt;" + Name</em>Of<em>TheElement + "&gt;"</em> + <em>Description of element's purpose</em>. For example, the <code><a href="/en-US/docs/Web/HTML/Element/video">&lt;video&gt;</a></code> element has a <em>title</em> of <em>&lt;video&gt;: The Video Embed element</em>.</p>
+<p>The frontmatter at the top of the page is used to define "page metadata". The values should be updated appropriately for the particular element.</p>
```
Not sure why it would need to be "page metadata".
```suggestion
<p>The frontmatter at the top of the page is used to define page metadata. The values should be updated appropriately for the particular element.</p>
```

Elchi3 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
+---</pre>
+
+<dl>
+<dt><strong>Title</strong></dt>
```
```suggestion
<dt><strong>title</strong></dt>
```

Elchi3 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
+    <p>Include the following tags as appropriate:</p>
+    <ul>
+      <li>Technology status: <strong>Experimental</strong> (if the technology is <a href="/en-US/docs/MDN/Guidelines/Conventions_definitions#experimental">experimental</a>), <strong>Deprecated</strong> (if it is <a href="/en-US/docs/MDN/Guidelines/Conventions_definitions#deprecated_and_obsolete">deprecated</a>).</li>
+      <li>Any other tags that represent terms people might search for related to the technology. For example the <code><a href="/en-US/docs/Web/HTML/Element/video">&lt;video&gt;</a></code> element includes the tags: <strong>HTML Video</strong>, <strong>Multimedia</strong>, <strong>Media Player</strong>, <strong>Movie Playback</strong>, etc.</li>
```
```suggestion
      <li>Any other tags that represent terms people might search for related to the technology. For example, the <code><a href="/en-US/docs/Web/HTML/Element/video">&lt;video&gt;</a></code> element includes the tags: <strong>HTML Video</strong>, <strong>Multimedia</strong>, <strong>Media Player</strong>, <strong>Movie Playback</strong>, etc.</li>
```

Elchi3 commented on files/en-us/mdn/structures/page_types/html_element_page_template/index.html:
```diff
@@ -5,51 +5,70 @@
 <div>\{{EmbedInteractiveExample("pages/tabbed/nameOfElement.html", "tabbed-standard")}}</div>
 
-<p>The source for this interactive example is stored in a GitHub repository. If you'd like to contribute to the interactive examples project, please clone <a href="https://github.com/mdn/interactive-examples">https://github.com/mdn/interactive-examples </a> and send us a pull request. <em>— THIS TEXT SHOULD BE HIDDEN USING THE "hidden" INLINE STYLE</em></p>
+<p>The source for this interactive example is stored in a GitHub repository. If you'd like to contribute to the interactive examples project, please clone <a href="https://github.com/mdn/interactive-examples">https://github.com/mdn/interactive-examples </a> and send us a pull request.</p>
```
Now we're not a wiki anymore and given we removed this text for the compat table, I think we could remove it for the interactive examples, too.

Elchi3 reviewed (changes requested):
Thanks Hamish, some comments.

hamishwillee committed:
Update files/en-us/mdn/structures/page_types/html_element_page_template/index.html

Co-authored-by: Florian Scholz <fs@florianscholz.com>

hamishwillee committed:
Update files/en-us/mdn/structures/page_types/html_element_page_template/index.html

Co-authored-by: Florian Scholz <fs@florianscholz.com>

hamishwillee committed:
Update files/en-us/mdn/structures/page_types/html_element_page_template/index.html

Co-authored-by: Florian Scholz <fs@florianscholz.com>

hamishwillee committed:
Update files/en-us/mdn/structures/page_types/html_element_page_template/index.html

Co-authored-by: Florian Scholz <fs@florianscholz.com>

hamishwillee committed:
Remove draft macro from page templates

hamishwillee committed:
Merge branch 'main' into page_template_html_element

hamishwillee committed:
Fix up case of Title in template front matter

hamishwillee commented:
Thanks! I've accepted suggestions, removed draft macro from all the templates, fixed up copy paste error in front matter title for all templates. Removed the interactive example text.
Should be good.

Elchi3 reviewed (approved):
Great, thank you! 👍
